tests/test_collections.py

```diff
@@ -40,61 +53,111 @@ def upload_file(self, contents):
-        contents = [col_file_item] * 8 + [col_ptr_item] * 8
 
-        with self.subTest("PUT new collection"):
-            res = self.app.put("/v1/collections",
-                               headers=get_auth_header(authorized=True),
-                               params=dict(uuid=str(uuid4()), version=datetime.now().isoformat(), replica="aws"),
-                               json=dict(name="n", description="d", details={}, contents=contents))
-            res.raise_for_status()
-            uuid, version = res.json()["uuid"], res.json()["version"]
-        with self.subTest("GET created collection"):
-            res = self.app.get("/v1/collections/{}".format(uuid),
-                               headers=get_auth_header(authorized=True),
-                               params=dict(version=version, replica="aws"))
-            res.raise_for_status()
-            self.assertEqual(res.json()["contents"], [col_file_item, col_ptr_item])
-        with self.subTest("GET latest version of collection"):
-            res = self.app.get("/v1/collections/{}".format(uuid),
-                               headers=get_auth_header(authorized=True),
-                               params=dict(replica="aws"))
-            res.raise_for_status()
-            self.assertEqual(res.json()["contents"], [col_file_item, col_ptr_item])
-        res = self.app.get("/v1/collections/{}".format(uuid),
+        res = self.app.put('/v1/files/' + file_uuid,
+                           params=dict(version=version),
+                           json=dict(creator_uid=0, source_url=source_url))
+        return file_uuid, res.json()["version"]
+
+    def _put(self, contents: typing.List,
+             authorized: bool=True,
+             uuid: typing.Optional[str]=None,
+             version: typing.Optional[str]=None,
+             replica: str='aws') -> typing.Tuple[str, str]:
+        uuid = str(uuid4()) if uuid is None else uuid
+        version = datetime.now().isoformat() if version is None else version
+
+        params = dict()
+        if uuid is not 'missing':
+            params['uuid'] = uuid
+        if version is not 'missing':
+            params['version'] = version
+        if replica is not 'missing':
+            params['replica'] = replica
+
+        res = self.app.put("/v1/collections",
+                           headers=get_auth_header(authorized=authorized),
+                           params=params,
+                           json=dict(name="n", description="d", details={}, contents=contents))
+        res.raise_for_status()
+        return res.json()["uuid"], res.json()["version"]
+
+    def _delete_collection(self, uuid: str, replica: str='aws'):
+        self.app.delete("/v1/collections/{}".format(uuid),
+                        headers=get_auth_header(authorized=True),
+                        params=dict(replica=replica))
+
+
+class Collections(unittest.TestCase, CollectionBase):
```

Comment:
What's the reason for breaking the naming convention for test cases?

Reply:
It's shorter, but I'll change it back to our general convention.